Sources/Aperture/Aperture.swift

```diff
@@ -1,228 +1,823 @@
+			assistant = AVOutputSettingsAssistant(
+				preset: .hevc7680x4320
+			)
+
+			assistant?.sourceVideoFormat = try? CMVideoFormatDescription(
+				videoCodecType: .hevc,
+				width: Int(dimensions.width),
+				height: Int(dimensions.height)
+			)
+		default:
+			assistant = nil
+		}
+
+		var outputSettings: [String: Any] = assistant?.videoSettings ?? [AVVideoCodecKey: options.videoCodec.asAVVideoCodec]
+
+		outputSettings[AVVideoWidthKey] = dimensions.width
+		outputSettings[AVVideoHeightKey] = dimensions.height
+
+		switch options.videoCodec {
+		case .h264:
+			outputSettings[AVVideoColorPropertiesKey] = [
+				AVVideoTransferFunctionKey: AVVideoTransferFunction_ITU_R_709_2,
+				AVVideoColorPrimariesKey: AVVideoColorPrimaries_ITU_R_709_2,
+				AVVideoYCbCrMatrixKey: AVVideoYCbCrMatrix_ITU_R_709_2
+			]
+		case .hevc:
+			outputSettings[AVVideoColorPropertiesKey] = [
+				AVVideoTransferFunctionKey: AVVideoTransferFunction_ITU_R_709_2,
+				AVVideoColorPrimariesKey: AVVideoColorPrimaries_P3_D65,
+				AVVideoYCbCrMatrixKey: AVVideoYCbCrMatrix_ITU_R_709_2
+			]
+		default:
+			break
+		}
+
+		let videoInput = AVAssetWriterInput(mediaType: .video, outputSettings: outputSettings)
+
+		self.videoInput = videoInput
+
+		videoInput.expectsMediaDataInRealTime = true
+
+		if assetWriter.canAdd(videoInput) {
+			assetWriter.add(videoInput)
+		} else {
+			continuation.resume(throwing: Aperture.Error.couldNotStartStream(Aperture.Error.couldNotAddInput("video")))
+			self.continuation = nil
+			return
+		}
+
+		isRunning = assetWriter.startWriting()
+		if isRunning {
+			assetWriter.startSession(atSourceTime: sampleBuffer.presentationTimeStamp)
+			continuation.resume(returning: ())
+		} else {
+			continuation.resume(throwing: Aperture.Error.couldNotStartStream(assetWriter.error))
+		}
+
+		self.continuation = nil
 	}
 
-	public func fileOutput(_ output: AVCaptureFileOutput, didResumeRecordingTo fileURL: URL, from connections: [AVCaptureConnection]) {
-		shouldPreventSleep = true
-		onResume?()
+	private func handleBuffer(buffer: CMSampleBuffer, isVideo: Bool) -> CMSampleBuffer {
+		/// Use the video buffer to syncronize after pausing if we have it, otherwise the audio
+		if !isVideo && target != .audioOnly {
+			return buffer
+		}
+
+		var resultBuffer = buffer
+
+		if isResuming {
+			isResuming = false
+
+			var pts = CMSampleBufferGetPresentationTimeStamp(buffer)
+			guard let lastFrame else {
+				return buffer
+			}
+			if lastFrame.flags.contains(.valid) {
+				if timeOffset.value > 0 {
+					pts = CMTimeSubtract(pts, timeOffset)
+				}
+
+				let offset = CMTimeSubtract(pts, lastFrame)
+				if timeOffset.value == 0 {
+					timeOffset = offset
+				} else {
+					timeOffset = CMTimeAdd(timeOffset, offset)
+				}
+			}
+			self.lastFrame?.flags = []
+		}
+
+		if timeOffset.value > 0 {
+			resultBuffer = resultBuffer.adjustTime(by: timeOffset) ?? resultBuffer
+		}
+
+		var lastFrame = CMSampleBufferGetPresentationTimeStamp(resultBuffer)
```

Comment:
Use `sampleBuffer` instance methods, not the legacy global methods. Applies in many places.